lago/verify_configuration.py

```diff
@@ -0,0 +1,399 @@
+#
+# Copyright 2014 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+#
+# Refer to the README and COPYING files for full details of the license
+#
+
+import os
+import commands
+import argparse
+import sys
+import getpass
+import platform
+import logging
+
+from .utils import LagoUserException
+
+LOGGER = logging.getLogger(__name__)
+
+class VerifyLagoStatus(object):
+    """
+    Verify Lago configuration
+    """
+    verificationStatus = False
+    def __init__(self,username,envs_dir,config_dict,verify_status):
+        self.username = username
+        self.envs_dir = envs_dir
+        self.groups = config_dict['groups']
+        self.nested = config_dict['nested']
+        self.virtualization = config_dict['virtualization']
+        self.lago_env_dir = config_dict['lago_env_dir']
+        self.kvm_configure = config_dict['kvm_configure']
+        self.install_pkg = config_dict['install_pkg']
+        self.home_permissions = config_dict['home_permissions']
+        self.ipv6_networking = config_dict['ipv6_networking']
+        VerifyLagoStatus.verificationStatus = verify_status
+
+    def displayLagoStatus(self,display_list):
+        """
+        Display Lago configuration status (OK/Not-OK) Verify Lago configuration
+        """
+        if (display_list):
+            print "Configuration Status:"
```

Comment:
A new verification step will require adding code in 4 different functions. It will be better to encapsulate all the information needed for a verification step inside an object.